var/spack/repos/builtin/packages/py-pennylane/package.py

```diff
@@ -36,8 +38,8 @@ class PyPennylane(PythonPackage):
     depends_on("py-semantic-version@2.7:", type=("build", "run"))
     depends_on("py-autoray@0.3.1:", type=("build", "run"))
     depends_on("py-cachetools", type=("build", "run"))
-    depends_on("py-pennylane-lightning@0.31.0:", type=("build", "run"), when="@0.31.0:")
-    depends_on("py-pennylane-lightning@0.30.0:", type=("build", "run"), when="@0.30.0:")
+    for v in range(30, 33):
+        depends_on(f"py-pennylane-lightning@0.{v}.0:", type=("build", "run"), when=f"@0.{v}.0:")
     depends_on(
         "py-pennylane-lightning@0.28.0:0.29.0", type=("build", "run"), when="@0.28.0:0.29.1"
     )
```

Comment:
Missing dep on typing-extensions

Reply:
Good catch, seems like it was introduced in v0.32 via the `dataset` module. I'll add it.